### PR TITLE
Fix some typing issues

### PR DESCRIPTION
**Pull request by alexfikl**
While playing with #235 and #237 to fix the mypy issues, I had some more random fixes sitting around.

This essentially removes the `ignore_errors = True` in `setup.cfg` for some modules and fixes the errors that popped up. There's a bunch of stuff in `persistent_dict.py` that probably requires a careful look.

cc @matthiasdiener since you typed most of `PersistentDict`.

### CODE REVIEW

File: pytools/mpiwrap.py [hunk 1]
```diff
@@ -13,7 +13,8 @@
 pytools.prefork.enable_prefork()
 
 
-if Is_initialized():  # noqa pylint:disable=undefined-variable
+# pylint:disable=undefined-variable
```
Comment (inducer) [inline pytools/mpiwrap.py:16]: I *think* this disables that warning for the whole file. Is that the intention?

**Review by inducer — COMMENTED** [review]
Found one small wrinkle. Otherwise LGTM.

File: pytools/mpiwrap.py (hunk 1 above)
Comment (alexfikl) [reply pytools/mpiwrap.py:16]: My reading of their documentation
https://pylint.pycqa.org/en/latest/user_guide/messages/message_control.html#block-disables
is that if there's a statement on the next line, it disables it for that, and if there's not it disables it for the whole block.

I can put it back on the same line, it just looked a bit silly with 3 types of ignores :(

File: pytools/mpiwrap.py [hunk 2]
```diff
@@ -13,7 +13,8 @@
...
+if Is_initialized():    # type: ignore[name-defined,unused-ignore] # noqa
```
Comment (alexfikl) [inline pytools/mpiwrap.py:17]: I'm not getting the `named-defined` error locally, but it shows up on the CI, so I added the `unused-ignore`, which is not great :(

File: pytools/mpiwrap.py (hunk 1 above)
Comment (inducer) [reply pytools/mpiwrap.py:16]: Did you mean `disable-next`?

Comment (alexfikl) [reply pytools/mpiwrap.py]: Yep :facepalm: Somehow I assumed that it just worked™ with the normal disable..

**Pull request merged**